src/nac/subviews/device_management.py

```diff
@@ -38,8 +40,18 @@ def get_queryset(self):
             device_list = device_list.filter(appl_NAC_DeviceRoleProd__in=selected_device_roles_prod)
         return device_list.order_by("name")
 
+    def get(self, request, *args, **kwargs):
+        # Check if the request is an AJAX request
+        if request.headers.get('X-Requested-With') == 'XMLHttpRequest':
+            # Handle AJAX request by rendering only the relevant part of the template
+            html = render_to_string('devices_results.html', {"device_list": self.get_queryset()})
```

Comment:
devices_results.html seems to be missing in the git repository. It would be good to have a test, to check if this site can be rendered ;)